homeassistant/components/home_connect/select.py

```diff
@@ -59,24 +73,31 @@ async def async_setup_entry(
 class HomeConnectProgramSelectEntity(HomeConnectEntity, SelectEntity):
     """Select class for Home Connect programs."""
 
+    entity_description: HomeConnectSensorEntityDescription
+
     def __init__(
         self,
         coordinator: HomeConnectCoordinator,
         appliance: HomeConnectApplianceData,
-        desc: SelectEntityDescription,
+        desc: HomeConnectSensorEntityDescription,
     ) -> None:
         """Initialize the entity."""
         super().__init__(
             coordinator,
             appliance,
             desc,
         )
+        self.start_on_select = desc.key == EventKey.BSH_COMMON_ROOT_ACTIVE_PROGRAM
```

Comment:
Side note: I'd replace this logic with a declarative approach by adding a couple of attributes to the entity description, eg `select_method` and `select_translation_key`. Then we don't need any conditions in `async_select_option` but can just use these attributes from the entity description. The `select_method` attribute can hold a lambda function taking the client instance as parameter.